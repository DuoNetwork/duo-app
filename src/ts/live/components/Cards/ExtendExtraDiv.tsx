import * as React from 'react';
import * as CST from '../../common/constants';
import { ColorStyles } from '../../common/styles';
import { SCardExtendExtraDiv } from './_styled';

interface IProps {
	accountShow: string;
	account: string;
	locale: string;
}

export default class ExtendExtraDiv extends React.Component<IProps> {
	public render() {
		const { account, accountShow, locale } = this.props;
		return (
			<SCardExtendExtraDiv
				color={accountShow === 'Unknown' ? ColorStyles.TextRedAlpha : undefined}
			>
				<div className="extend-extra-wrapper">
					<div className="tag-title">{CST.TH_ADDRESS[locale]}</div>
					<a
						className="tag-content"
						href={
							'https://' +
							(__KOVAN__ ? 'kovan.' : '') +
							'etherscan.io' +
							(account ? '/address/' + account : '')
						}
						target="_blank"
					>
						{accountShow}
					</a>
				</div>
			</SCardExtendExtraDiv>
		);
	}
}
