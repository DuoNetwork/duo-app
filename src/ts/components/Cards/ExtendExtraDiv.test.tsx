// fix for @ledgerhq/hw-transport-u2f 4.28.0
import '@babel/polyfill';
import { shallow } from 'enzyme';
import * as React from 'react';
import ExtendExtraDiv from './ExtendExtraDiv';

describe('ExtendExtraDiv Test', () => {
	describe('Test Snapshot', () => {
		const accountShow = "123";
		const account = '0x0';
		const locale = 'EN';

		it('Test Snapshot', () => {
			const wrapper = shallow(<ExtendExtraDiv accountShow={accountShow} account={account} locale={locale} />);
			expect(wrapper).toMatchSnapshot();
		});
	});
});
