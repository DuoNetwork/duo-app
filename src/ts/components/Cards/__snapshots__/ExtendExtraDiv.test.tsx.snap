// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExtendExtraDiv Test Test Snapshot Test Snapshot 1`] = `
<styled.div>
  <div
    className="extend-extra-wrapper"
  >
    <div
      className="tag-title"
    >
      Address
    </div>
    <a
      className="tag-content"
      href="https://kovan.etherscan.io/address/0x0"
      target="_blank"
    >
      123
    </a>
  </div>
</styled.div>
`;
